docs/usage/manifest.md

```diff
@@ -0,0 +1,84 @@
+                                 infoPlist: "Config/App-Info.plist",
+                                 sources: ["Sources/**"],
+                                 resources: ["Resources/**"],
+                                 dependencies: [
+                                     .project(target: "Framework1", path: "../Framework1"),
+                                     .project(target: "Framework2", path: "../Framework2"),
+                                 ])
+])
+```
+
+### Project
+
+A `Project.swift` should initialize a variable of type `Project.swift`. It can take any name, although we recommend to stick to `project`. A project accepts the following attributes:
+
+- **Name:** Name of the project. It's used to determine the name of the generated Xcode project.
+- **Targets:** It contains the list of targets that belong to the project. Read more [about targets](#target)
+- **Settings (optional):** Read more about [settings](#settings)
+
+> **Note:** All the relative paths in the project manifest are relative to the folder that contains the manifest file.
+
+### Target
+
+Each target in the list of project targets can be initialized with the following attributes:
+
+- **Name:** The name of the target. The Xcode project target and the derivated product take the same name.
+- **Platform:** The platform the target product is build for. The following products are supported:
+  - **.app:** An application
+  - **.staticLibrary:** A static library.
+  - **.dynamicLibrary:** A dynamic library.
+  - **.framework:** A dynamic framework.
+  - **.unitTests:** A unit tests bundle.
+  - **.uiTests:** A UI tests bundle.
+- **BundleID:** The product bundle identifier.
+- **InfoPlist:** Relative path to the `Info.plist`.
+- **Sources:** List of sources to be compiled by the target. The attribute can be any of the following types:
+
+  - **String:** A file or glob pattern _(e.g. `Sources/**`)_
+  - **[String]:** A list of files or list of glob patterns _(e.g. `["Sources/**"]`)_
+
+- **Resources (optional):** List of resources to be included in the product bundle. The types that it can take are the same as the `sources` attribute.
+- **Headers (optional):** Target headers. It accepts a `Header` type that is initialized with the following attributes:
+
+  - **Public:** Relative path to the folder that contains the public headers.
+  - **Private:** Relative path to the folder that contains the private headers.
+  - **Project:** Relative path to the folder that contains the project headers.
+
+- **Entitlements (optional):** Relative path to the entitlements file.
+- **Actions (optional):** Target actions allow defining extra script build phases. It's an array of `TargetAction` objects that can be initialized with any of the following static functions:
+
+  - **pre(tool:):** Runs the binary with the given name before the default target build phases.
+  - **pre(path:):** Runs the binary at the given path before the default target build phases.
```

Comment:
if I'm not mistaken the use case is:
`.pre(tool: "/bin/sh", path: "my_script.sh")`  (default is `/bin/sh`)

Is it worth for this one including examples to clarify what is meant by binary?

`.pre(...)`: Runs before the target's default build phase
`.post(...)`: Runs before the target's default build phase

Parameters:
- **tool (optional)**: The binary to use to execute the build phase.
- **path (optional)**: Path to the script to execute.
- **arguments**: Path to the script to execute.
- **name**: The name of the build phase

Examples:

`.pre(path: "my_custom_script.sh", arguments: ["hello"], name: "My Custom Script Phase")`
...

Reply:
Good idea 👍